File: sort-array-by-parity.cpp
/*
 *Solution for problem 905
 *https://leetcode.com/problems/sort-array-by-parity/
 */
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        vector<int> tmp;
        stack<int> s;
        for(int i=0; i<A.size(); i++){
            if(A[i] % 2 ==1){
                s.push(A[i]);
            }
            else{
                tmp.push_back(A[i]);
            }
        }
        while(!s.empty()){
            tmp.push_back(s.top());
            s.pop();
        }
        return tmp;
    }
};

File: solutions/sort-array-by-parity.py
"""
905. Sort Array By Parity
- https://leetcode.com/problems/sort-array-by-parity/
- https://leetcode.com/explore/challenge/card/august-leetcoding-challenge/551/week-3-august-15th-august-21st/3431/
"""
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        oddList = []
        evenList = []
        for x in A:
            if x % 2 == 0:
                evenList.append(x)
            else:
                oddList.append(x)
        return evenList + oddList
